File: checkers/217/main.cpp
#include <checkers/testlib.h>
#include <iostream>
#include <vector>
#include <set>
#include <algorithm>
#include <map>
#include <set>
#include <string>
#include <sstream>
#include <regex>

using namespace NTestlib;

int main(int argc, char *argv[])
{
    InitChecker(argc, argv);
    std::vector<bool> a;
    std::vector<std::string> b;
    while (File.HasInput())
        b.push_back(File.ReadLine());
    int cnt1 = 0;
    while (Ans.HasInput())
    {
        int x = Ans.ReadInt();
        if (x)
            cnt1++;
        a.push_back((x));
    }
    int cnt = 0;
    while (Out.HasInput())
    {
        int x = Out.ReadInt(1, a.size());
        int y = Out.ReadInt(1, b[x - 1].length() - 13);
        if (!a[x - 1])
            QuitWith(WA, "Merlin was silent");
        if (b[x - 1].substr(y - 1, 14) != "Avada-ke-davra")
            QuitWith(WA, "Merlin didn't say that");
        cnt++;
    }
    if (cnt == cnt1)
        QuitWith(AC, "OK");
    else
        QuitWith(WA, "Merlin said more");
}

File: checkers/217/main.go
package main

import (
	"fmt"
	"testlib"
	"testlib/core"
	"testlib/utils"
)

func i2b(i_0 int) bool {
	if i_0 == 0 {
		return false
	} else {
		return true
	}
}

func createMatrix0(n int, m int) [][]int {
	matrix := make([][]int, n)
	for i := 0; i < n; i++ {
		matrix[i] = make([]int, m)
		for j := 0; j < m; j++ {
			matrix[i][j] = 0
		}
	}
	return matrix
}

func checkValueOutOfSegmen(val int, a int, b int) bool {
	if val < a || val > b {
		return true
	} else {
		return false
	}
}

func main() {
	defer func() {
		_ = testlib.GetStream(testlib.TaskOutput).Close()
		_ = testlib.GetStream(testlib.InputFromUser).Close()
	}()
	inputStream := testlib.GetReader(testlib.TaskInput)
	outputStream := testlib.GetReader(testlib.TaskOutput)
	inputFromUserStream := testlib.GetReader(testlib.InputFromUser)

	a := make([]bool, 0)
	b := make([]string, 0)
	for {
		if flag, err := utils.IsStreamFinish(inputStream, inputStream); flag {
			if err != nil {
				testlib.WriteAnswer(err.Error(), testlib.EF)
				return
			}
			break
		}
		b = append(b, core.ReadLine(inputStream, true))
	}
	cnt1 := 0
	x := false
	for {
		if flag, err := utils.IsStreamFinish(outputStream, outputStream); flag {
			if err != nil {
				testlib.WriteAnswer(err.Error(), testlib.EF)
				return
			}
			break
		}
		x = i2b(core.ReadInt(outputStream))
		if x {
			cnt1++
		}
		a = append(a, x)
	}
	cnt := 0
	xint := 0
	y := 0
	for {
		if flag, err := utils.IsStreamFinish(inputFromUserStream, inputFromUserStream); flag {
			if err != nil {
				testlib.WriteAnswer(err.Error(), testlib.WA)
				return
			}
			break
		}
		xint = core.ReadInt(inputFromUserStream)
		if checkValueOutOfSegmen(xint, 1, len(a)) {
			testlib.WriteAnswer(fmt.Sprintf("out of segment xint(%d) [1,len(a)(%d)]", xint, len(a)), testlib.WA)
			return
		}
		y = core.ReadInt(inputFromUserStream)
		if checkValueOutOfSegmen(y, 1, len(b[xint-1])-13) {
			testlib.WriteAnswer(fmt.Sprintf("out of segment y(%d) [1,len(a)(%d)]", y, len(b[xint-1])-13), testlib.WA)
			return
		}
		if !a[xint-1] {
			testlib.WriteAnswer(fmt.Sprintf("Merlin was silent"), testlib.WA)
			return
		}
		if b[xint-1][y-1:14] != "Avada-ke-davra" {
			testlib.WriteAnswer(fmt.Sprintf("Merlin didn't say that"), testlib.WA)
			return
		}
		cnt++
	}
	if cnt == cnt1 {
		testlib.WriteAnswer(fmt.Sprintf("OK"), testlib.OK)
	} else {
		testlib.WriteAnswer(fmt.Sprintf("Merlin said more"), testlib.WA)
	}
}
